src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPump.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using CoreWCF.Configuration;
+using CoreWCF.Queue.Common;
+using CoreWCF.Runtime;
+using Microsoft.Extensions.Logging;
+
+namespace CoreWCF.Channels;
+
+internal sealed class KafkaTransportPump : QueueTransportPump, IDisposable
+{
+    private readonly ILogger<KafkaTransportPump> _logger;
+    private readonly KafkaDeliverySemantics _kafkaDeliverySemantics;
+    internal IConsumer<Null, byte[]> Consumer { get; private set; }
+    internal ConsumerConfig ConsumerConfig { get; private set; }
+    internal IProducer<Null, byte[]> Producer { get; private set; }
+    internal string Topic { get; }
+    internal KafkaTransportBindingElement TransportBindingElement { get; }
+    private CountdownEvent _receiveContextCountdownEvent;
+
+    private readonly Uri _baseAddress;
+    private CancellationTokenSource _cts;
+    private AsyncManualResetEvent _mres;
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atMostOnceConfigValues = (false, null);
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atLeastOncePerMessageCommitConfigValues = (false, null);
+    private static readonly (bool? EnableAutoCommit, bool? EnableAutoOffsetStore) s_atLeastOnceBatchCommitConfigValues = (true, false);
+    private static readonly Regex s_topicNameRegex =
+        new(@"^[a-zA-Z0-9\.\-_]{1,255}$", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
+
+    public KafkaTransportPump(KafkaTransportBindingElement transportBindingElement,
+        ILogger<KafkaTransportPump> logger,
+        IServiceDispatcher serviceDispatcher, KafkaDeliverySemantics kafkaDeliverySemantics)
+    {
+        _logger = logger;
+        _kafkaDeliverySemantics = kafkaDeliverySemantics;
+        Topic = serviceDispatcher.BaseAddress.PathAndQuery.TrimStart('/');
+        if (string.IsNullOrEmpty(Topic) || !s_topicNameRegex.IsMatch(Topic))
+        {
+            throw new NotSupportedException(string.Format(SR.InvalidTopicName, Topic));
+        }
+
+        if (transportBindingElement.ErrorHandlingStrategy == KafkaErrorHandlingStrategy.DeadLetterQueue)
+        {
+            if (string.IsNullOrEmpty(transportBindingElement.DeadLetterQueueTopic))
+            {
+                throw new NotSupportedException(SR.InvalidDeadLetterQueueTopicName);
+            }
+
+            if (!s_topicNameRegex.IsMatch(transportBindingElement.DeadLetterQueueTopic))
+            {
+                throw new NotSupportedException(string.Format(SR.InvalidTopicName, Topic));
+            }
+        }
+        TransportBindingElement = transportBindingElement;
+        _baseAddress = serviceDispatcher.BaseAddress;
```

Comment:
If you are going to keep this around, you could have problems if the passed in binding element is modified by poorly written code. You should clone the binding element that was passed in to protect against this.

Reply:
done